include/pybind11/cast.h

```diff
@@ -114,44 +114,38 @@ PYBIND11_NOINLINE inline internals &get_internals() {
 /// A life support system for temporary objects created by `type_caster::load()`.
 /// Adding a patient will keep it alive up until the enclosing function returns.
 class loader_life_support {
-    /// A list of patiences which is created when a function is first entered
-    /// and destroyed after it returns (nested calls share the same list).
-    static PyObject *&patients() {
-        static PyObject *ptr = nullptr;
-        return ptr;
-    }
-
-    /// Nested calls only increment/decrement the level
-    static int &level() {
-        static int value = 0;
-        return value;
-    }
-
 public:
+    /// A list of patiences is created when a function is entered
     loader_life_support() {
-        if (level() == 0)
-            patients() = PyList_New(0);
-        ++level();
+        get_internals().loader_patient_stack.push_back(PyList_New(0));
```

Comment:
This seems like it's going to be creating a lot of lists here, the vast majority of which are never used (i.e. life support is the exception rather than the rule).  Is there a reason to prefer always allocating here to allocating the list on demand when the first `add_patient` is called?